datalad_next/iter_collections/tests/test_itergitworktree.py

```diff
@@ -0,0 +1,74 @@
+from pathlib import PurePath
+from ..gitworktree import (
+    GitWorktreeItem,
+    GitWorktreeFileSystemItem,
+    iter_gitworktree,
+)
+
+
+def test_iter_gitworktree(existing_dataset):
+    ds = existing_dataset
+
+    (ds.pathobj / 'emptydir').mkdir()
+    untracked = ds.pathobj / 'subdir' / 'untracked'
+    untracked.parent.mkdir()
+    untracked.write_text('untracked')
+
+    tracked_items = list(iter_gitworktree(ds.pathobj, untracked=None))
+    # without untracked's and no link resolution this is plain and fast
+    assert all(
+        isinstance(i, GitWorktreeItem) and i.gitsha and i.gittype
+        for i in tracked_items
+    )
+
+    all_items = list(iter_gitworktree(ds.pathobj, untracked='all'))
+    # empty-dir is not reported, only untracked files
+    assert len(all_items) == len(tracked_items) + 1
+    assert any(
+        i.name == PurePath('subdir', 'untracked')
```

Comment:
Must be PurePosixPath. Otherwise it will fail on Windows (because Windows will return a `PureWindowsPath`-instance if PurePath() is called).

```suggestion
        i.name == PurePosixPath('subdir', 'untracked')
```